### DISCUSSION TITLE
how to send file body, like file upload

### DISCUSSION THREAD

**Discussion by xiaoliu10** [ledgetech/lua-resty-http#258] [Q&A]
I want to send files, but I can't find examples or documents
like send local file to other server

**Comment by xiaoliu10** [comment]
Can someone help me

**Comment by pintsized** [comment]
It depends on what you mean by "local file", because typically in OpenResty there is not a safe way to read file data from disk on the fly (it would block Nginx). You would need to read any file data into Lua in advance, during the `init` phase (i.e when Nginx starts), which may not be what you want. Again though, this is outside of the scope of this project. You would be better off asking for help on the OpenResty mailing lists or similar.

If you have file data in Lua already, and you can figure out how to make the required HTTP request for your use case, then yes, this library is the tool you would use to achieve that.

**Comment by xiaoliu10** [comment]
Send a file, such as a local file, read the file stream, and then send it to another server

**Reply by liweinan** [reply]
I have the similar usecase, and I can't find example online, can anyone provide a usage example on file uploading(as client side, to upload a file from local filepath to a remote url)?
